File: Python/Pattern wise question/BFS/11_right_view.py
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        if(not root):
            return res
        q = deque()
        q.append(root)

        while(q):
            sz = len(q)
            for i in range(sz):
                curr = q.popleft()
                if(i == sz-1):
                    res.append(curr.val)
                if(curr.left):
                    q.append(curr.left)
                if(curr.right):
                    q.append(curr.right)
        return res
        

File: C++/Pattern wise question/BFS/11_right_view.cpp

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> res;
        if(!root) return res;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            int sz = q.size();
            for(int i=0;  i<sz; i++){
                TreeNode* temp = q.front();
                q.pop();
                if(i == sz-1) res.push_back(temp->val);

                if(temp->left) q.push(temp->left);

                if(temp->right) q.push(temp->right);
            }
        }
        return res;
    }
};
